### ISSUE TITLE
08-CMSIS: stack frame(r7) is not correct: r7 will be pushed to MSP when going into systick interrupt, and r7 will be poped from MSP when back from systick interrupt

### ISSUE THREAD

**Issue by bbbluesky** [jserv/mini-arm-os#37]
r7 will be pushed to MSP when going into systick interrupt, and r7 will be poped from MSP when back from systick interrupt
![image](https://github.com/user-attachments/assets/d7c008cf-8d16-42e1-aa1f-f92f43c8bee7)
for example, currently task1 is running, then systick interrupt comes, and r7(task1 stack frame) will be pushed to MSP at the same time. After coming back from pendSV, the r7(task1 stack frame) will be poped from MSP, while another task is runing(not task1), which means the r7 is not correct.

**Comment by lecopzer** [comment]
What's issue you are facing with?
Why systick coming back from pendSV? PendSV is the lowest priority, pendSV will never interrupt systick, what's condition you had to get this happened?

**Comment by bbbluesky** [comment]
It is OK now. I ignored that PendSV should be the lowest priority. So the system was set to default priority: Systick is 15, while PendSV is 0. If the PendSV is priorier than systick, PendSV will triggered not until systick finished.
Thanks for replying
